frontend/src/lib/requests.ts

```diff
@@ -0,0 +1,34 @@
+import { API_BASE_URL, log } from '.';
+
+import type { AxiosRequestConfig } from 'axios';
+import axios from 'axios';
+
+const quizootApi = axios.create({
+    baseURL: API_BASE_URL,
+    timeout: 1000,
+    maxBodyLength: 2000,
+});
+
+export type ApiResponse =
+    | number
+    | string
+    | boolean
+    | null
+    | ApiResponse[]
+    | { [x: string]: ApiResponse };
+
+export async function makeRequest<TData extends ApiResponse>(
+    config: AxiosRequestConfig
```

Comment:
Isn't `request` a better name than `config` ?

Reply:
I basically used the same prototype used by axios itself:

```typescript
<any, AxiosResponse<any, any>, any>(config: AxiosRequestConfig<any>) => Promise<AxiosResponse<any, any>>

// The whole interface is as follow in frontend/node_modules/axios/index.d.ts:
export interface AxiosInstance extends Axios {
  <T = any, R = AxiosResponse<T>, D = any>(config: AxiosRequestConfig<D>): Promise<R>;
  <T = any, R = AxiosResponse<T>, D = any>(url: string, config?: AxiosRequestConfig<D>): Promise<R>;

  defaults: Omit<AxiosDefaults, 'headers'> & {
    headers: HeadersDefaults & {
      [key: string]: AxiosHeaderValue
    }
  };
}
```

And as the name `AxiosRequestConfig` is suggesting, it is a configuration. Does that convince you ? We can name it `requestConfig` if you want

Reply:
`requestConfig` is Ok